pkg/resources/validation/cluster/cluster.go

```diff
@@ -115,7 +115,19 @@ func (v *Validator) validateFleetPermissions(request *admission.Request) (*admis
 		return nil, fmt.Errorf("failed to get old and new clusters from request: %w", err)
 	}
 
-	if newCluster.Spec.FleetWorkspaceName == "" || oldCluster.Spec.FleetWorkspaceName == newCluster.Spec.FleetWorkspaceName {
+	// Ensure that the FleetWorkspaceName field cannot be unset, as it would cause (likely unintentional) cluster
+	// deletion.
+	if newCluster.Spec.FleetWorkspaceName == "" && oldCluster.Spec.FleetWorkspaceName != "" {
+		return &admissionv1.AdmissionResponse{
+			Result: &metav1.Status{
+				Status:  "Failure",
+				Message: "field FleetWorkspaceName cannot be unset",
+				Reason:  metav1.StatusReasonInvalid,
+				Code:    http.StatusBadRequest,
+			},
+			Allowed: false,
+		}, nil
+	} else if oldCluster.Spec.FleetWorkspaceName == newCluster.Spec.FleetWorkspaceName {
 		return &admissionv1.AdmissionResponse{
 			Allowed: true,
 		}, nil
```

Comment:
Since we are returning in the above condition, we do not need an else statement.
```suggestion
	if newCluster.Spec.FleetWorkspaceName == "" && oldCluster.Spec.FleetWorkspaceName != "" {
		return &admissionv1.AdmissionResponse{
			Result: &metav1.Status{
				Status:  "Failure",
				Message: "field FleetWorkspaceName cannot be unset",
				Reason:  metav1.StatusReasonInvalid,
				Code:    http.StatusBadRequest,
			},
			Allowed: false,
		}, nil
	}
	if oldCluster.Spec.FleetWorkspaceName == newCluster.Spec.FleetWorkspaceName {
		return admission.ResponseAllowed(), nil
```